compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt

```diff
@@ -65,7 +64,13 @@ internal class MouseWheelScrollNode(
         }
     }
 
-    private val pointerInputNode = delegate(SuspendingPointerInputModifierNode {
+    // Note that when `MouseWheelScrollNode` is used as a delegate of `ScrollableNode`,
+    // pointerInputNode does not get dispatched pointer events to in the standard manner because
+    // Modifier.Node.dispatchForKind does not dispatch to child/delegate nodes of the matching type,
+    // and `ScrollableNode` is already an instance of `PointerInputModifierNode`.
```

Comment:
> `ScrollableNode` is already an instance of `PointerInputModifierNode`

I guess it's a root of the problem. And the correct approach is to have two input modes (drag and wheel) at the same level inside other one like it was before. Having handling wheel inside handling drag doesn't seem good

Reply:
It seems like https://android-review.googlesource.com/c/platform/frameworks/support/+/2996341 "optimization" was not just aware of mouse changes in fork. I think it makes sense to partially revert to old approach